### DISCUSSION TITLE
Importing very large CIFs

### DISCUSSION THREAD

**Discussion by lnaglecocco** [materialsproject/pymatgen#2811] [Q&A] [answered]
Hi all, 

I am attempting to use Pymatgen for the analysis of a supercell containing ~10,000 atoms. I am getting the following warning message:
```
UserWarning: Issues encountered while parsing CIF: Some fractional co-ordinates rounded to ideal values to avoid issues with finite precision.
  warnings.warn("Issues encountered while parsing CIF: %s" % "\n".join(self.warnings))
```
I understand the rationale for including this feature, but in my specific circumstance this will harm my analysis. I think that the from_file() method should have an optional argument to switch this off, i.e. force it to accept the atomic positions in use. Is there any way to do this? Could someone add this feature?

Liam

**Comment by janosh** [comment]
@lnaglecocco Just out of curiosity, could you explain how this affects your analysis?

The relevant code is in `CifParser._sanitize_data`. If you want to add a flag to returns `data` early before rounding finite precision fractional coordinates, I'd be happy to take a PR for that.

https://github.com/materialsproject/pymatgen/blob/666f90f7e6244fbb1c1288239855fd2223e54126/pymatgen/io/cif.py#L543-L563

**Reply by lnaglecocco** [reply]
Hi @janosh , thank you for getting back to me so quickly.

I'm analysing the output of a big box refinement of neutron Pair Distribution Function data, running calculations for various parameters (bond lengths, bond length distortion index, Van Vleck mode calculations) using code I have written on top of pymatgen. My supercell is ~50 x ~50 x ~50 Angstroms with approx 10,000 atoms. If the positions of atomic sites are getting rounded during importing from a CIF to a pymatgen Structure object, then it seems to me that any subsequent calculations are going to be less precise. It's entirely possible I've misunderstood the error and perhaps this is a non-issue. If I've misunderstood, please let me know

The solution you describe sounds like it would help.

**Comment by shyuep** [comment, marked as answer]
I just pushed an option to allow this to be bypassed.

**Reply by lnaglecocco** [reply]
Hi Dr Ong,

Thank you for doing this. I can see you've made a change to CifParser with commit 8fa4dcd but in terms of the `Structure.from_file()` method I'm not sure how to fix your `frac_tolerance` parameter to a certain value. Would this require an additional commit to pymatgen.core.structure, to add an argument to this `.from_file()` method? 

Liam

**Reply by shyuep** [reply]
I just released a new version. Once you update to 2023.1.20, you should be able to use Structure.from_file(frac_tolerance=0)

**Reply by lnaglecocco** [reply]
Thank you!

**Reply by lnaglecocco** [reply]
Hi @shyuep ,
I've updated pymatgen and it is working as you said it would. Thanks again for your help, and for this incredibly speedy fix!
Liam
